### PR TITLE
test: 100% coverage of virtual-select component

### PR DESCRIPTION
**Pull request by suyashpatil78**
[workerB](https://workerb.linearb.io/v2/badge/collaboration-page?magicLinkId=5PuSBcy)
### <samp>🤖 Generated by Copilot at 56b2cf7</samp>

Added a shared component and unit tests for virtual select modal. The `VirtualSelectComponent` allows users to select an option from a list using a modal, and integrates with the `ReactiveFormsModule`. The `virtual-select.component.spec.ts` file contains the unit tests for the component's methods and properties. The `modal-controller.data.ts` file provides mock data for creating the modal.

### <samp>🤖 Generated by Copilot at 56b2cf7</samp>

> _We test the virtual select, the component of control_
> _We mock the modal data, the source of our console_
> _We write and register, the methods of the interface_
> _We open the modal gate, the portal to our fate_

### <samp>🤖 Generated by Copilot at 56b2cf7</samp>

*  Import and mock `VirtualSelectModalComponent` for testing `VirtualSelectComponent` ([link](https://github.com/fylein/fyle-mobile-app/pull/2500/files?diff=unified&w=0#diff-32f470962ca37f24fcdc6eaf809fb30dc2ce6a9e003b41b20b34908107e83805R22), [link](https://github.com/fylein/fyle-mobile-app/pull/2500/files?diff=unified&w=0#diff-91d85063a48a1368863ce0b7f7cdf1385918561b5a047d30255f021fae9bfaccR1-R237))
*  Add mock data object for creating virtual select modal using `ModalController` service ([link](https://github.com/fylein/fyle-mobile-app/pull/2500/files?diff=unified&w=0#diff-32f470962ca37f24fcdc6eaf809fb30dc2ce6a9e003b41b20b34908107e83805R448-R475))
*  Test `VirtualSelectComponent`'s creation, modal opening, value setting, validation, and `ControlValueAccessor` methods ([link](https://github.com/fylein/fyle-mobile-app/pull/2500/files?diff=unified&w=0#diff-91d85063a48a1368863ce0b7f7cdf1385918561b5a047d30255f021fae9bfaccR1-R237))

## Clickup
Please add link here

## Code Coverage
https://app.clickup.com/t/85zu3u05y

## UI Preview
Please add screenshots for UI changes

### CODE REVIEW

**Review by OmkarJ13 — APPROVED** [review]

File: src/app/shared/components/virtual-select/virtual-select.component.spec.ts [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+      selectionModalSpy.onWillDismiss.and.resolveTo({
+        data: {
+          value: 'value',
+        },
+      });
+      modalProperties.getModalDefaultProperties.and.returnValue(properties);
+      modalController.create.and.resolveTo(selectionModalSpy);
+    });
+
+    it('openModal(): should open select modal and set value equals to value returned by modal', fakeAsync(() => {
```
Comment (hlkavya0213) [inline src/app/shared/components/virtual-select/virtual-select.component.spec.ts:74]: if `openModal():` is added in describe no need to add again inside the it, fix it for all

File: src/app/shared/components/virtual-select/virtual-select.component.spec.ts [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+  });
+
+  describe('set value():', () => {
+    beforeEach(() => {
+      //@ts-ignore
+      component.innerValue = 'ECONOMY';
+      component.options = [
+        { label: 'Business', value: 'BUSINESS' },
+        { label: 'Economy', value: 'ECONOMY' },
+        { label: '', value: '' },
```
Comment (hlkavya0213) [inline src/app/shared/components/virtual-select/virtual-select.component.spec.ts:133]: when do we get the label as empty string ?

File: src/app/shared/components/virtual-select/virtual-select.component.spec.ts [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+      ];
+      //@ts-ignore
+      spyOn(component, 'onChangeCallback');
+    });
+
+    it('should set the value and update the displayValue if any option.value is equal to innerValue', () => {
```
Comment (hlkavya0213) [inline src/app/shared/components/virtual-select/virtual-select.component.spec.ts:139]: should set the value and update the displayValue if any "option.value" is equal to innerValue

File: src/app/shared/components/virtual-select/virtual-select.component.spec.ts [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+    });
+
+    it('should set the value and update the displayValue if type of innerValue is string', () => {
+      component.value = '';
+      //@ts-ignore
+      expect(component.innerValue).toEqual('');
+      expect(component.displayValue).toEqual('');
+      //@ts-ignore
+      expect(component.onChangeCallback).toHaveBeenCalledOnceWith('');
+    });
```
Comment (hlkavya0213) [inline src/app/shared/components/virtual-select/virtual-select.component.spec.ts:155]: one line space between each it()

File: src/app/shared/components/virtual-select/virtual-select.component.spec.ts [hunk 5]
```diff
@@ -0,0 +1,237 @@
...
+    it('should set the value and update the displayValue if innerValue and defaultLabelProps is defined', () => {
+      component.defaultLabelProp = 'vendor';
+      component.value = { travelClass: 'BUSINESS', vendor: 'vendor1' };
+      //@ts-ignore
+      expect(component.innerValue).toEqual({ travelClass: 'BUSINESS', vendor: 'vendor1' });
+      expect(component.displayValue).toEqual('vendor1');
+      //@ts-ignore
+      expect(component.onChangeCallback).toHaveBeenCalledOnceWith({ travelClass: 'BUSINESS', vendor: 'vendor1' });
+    });
+    it('should set the value and update the displayValue if none of the conditions holds true', () => {
```
Comment (hlkavya0213) [inline src/app/shared/components/virtual-select/virtual-select.component.spec.ts:165]: same as above, fix it for all

File: src/app/shared/components/virtual-select/virtual-select.component.spec.ts [hunk 6]
```diff
@@ -0,0 +1,237 @@
...
+      //@ts-ignore
+      component.innerValue = 'ECONOMY';
+      component.options = [
+        { label: 'Business', value: 'BUSINESS' },
+        { label: 'Economy', value: 'ECONOMY' },
+        { label: '', value: '' },
+      ];
+    });
+
+    it('should set the value and update the displayValue if any option.value is equal to innerValue', () => {
```
Comment (hlkavya0213) [inline src/app/shared/components/virtual-select/virtual-select.component.spec.ts:195]: should set the value and update the displayValue if any "option.value" is equal to innerValue

File: src/app/shared/components/virtual-select/virtual-select.component.spec.ts [hunk 7]
```diff
@@ -162,6 +163,7 @@ describe('VirtualSelectModalComponent', () => {
       //@ts-ignore
       expect(component.onChangeCallback).toHaveBeenCalledOnceWith({ travelClass: 'BUSINESS', vendor: 'vendor1' });
     });
+
```
Comment (suyashpatil78) [inline src/app/shared/components/virtual-select/virtual-select.component.spec.ts:166]: added one line space, don't know why github not showing as resolved

**Review by Chethan-Fyle — APPROVED** [review]

**Review by hlkavya0213 — APPROVED** [review]

**Pull request merged**